webcrawler.py

```diff
@@ -50,9 +59,10 @@ def collate():
     with open(SCRAPY_OUTPUT_FILE) as fd:
         results = json.load(fd)
     for result in results:
-        url = result['Link']
-        result_keywords[url].add(result['word'])
-        result_titles[url] = result['title']
+        if (contains_date(result['Link'])):
+            url = result['Link']
```

Comment:
Minor nitpick: since you're already making a variable for the URL, you might as well make that variable first and use it in the `if`.